### PR TITLE
Sitemap generation on pre-push hook

### PR DESCRIPTION
**Pull request by imjuangarcia**
## 📚 Context

This PR fixes an issue where sitemaps aren't generated properly on our `postbuild` script, by enforcing its generation on a pre-push hook.

## 🧠 Description of Changes

* Added a `pre_push` hook on `.husky` that will trigger the sitemap generation before your code gets pushed;
* Added a `sitemap` script on `package.json` that will build the site and run the sitemap generation afterwards

## 💥 Impact

Size:

- [x] Small 
- [ ] Not small 

## 🌐 References

- [ ] [Notion](...)

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

### CODE REVIEW

**Comment by imjuangarcia** [comment]
@snehankekre Please checkout the branch and give it a try! Let me know if any adjustments are needed 🙂

**Review by snehankekre — APPROVED** [review]
LGTM @imjuangarcia! Thank you! Approved pending the following clarification: is it possible to omit from the sitemap `/menu`, `/.keep`, `/style-guide`, `/gdpr-banner`, and `/index`? Possibly even `/library` as its supposed to redirect to `/library/get-started`.

Not sure if there's value in search engines crawling / indexing these pages. I will defer to you. 🙂

**Comment by imjuangarcia** [comment]
@snehankekre Done! Excluded those files in [0d7b65d](https://github.com/streamlit/docs/pull/420/commits/0d7b65d4300d9940b6022fa95a82b3e88d1e01a8)

**Pull request merged**
